### PR TITLE
Replaced single_task reduction to parallel reduction and updated sample as per latest coding guidelines

### PR DESCRIPTION
**Pull request by mpanoop**
# Replaced single_task reduction to parallel reduction and updated code to latest coding guidelines

1. Previously there was a third kernel which was exclusively doing serial reduction of the energy array whose memory footprint was "n" (number of particles) and this array was memory synchronized for iteration between CPU and GPU. Updated the sample to support parallel reduction and hence the third kernel is removed and the reduction operation is integrated to the second kernel. Also the memory footprint of the same is reduced to a scalar float instead of an array of "n" elements.

2. Accessor returned from a buffer using get_access is by default access::mode::read_write. So updated the sample to remove the explicit mention of this access mode.

3. Updated the README.md to reflect the parallel reduction feature in DPC++.

Fixes # (issue) 

## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X ] New feature (non-breaking change which adds functionality)
- [ ] Sample Migration (Moving sample from old repository after completing checklist established)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ X ] Command Line
- [ ] oneapi-cli
- [ ] Visual Studio
- [ ] Eclipse IDE
- [ ] VSCode

# Checklist for Moving samples:
Links and Details can be found in the samples WG Teams Files. 

- [ ] Review sample design with domain reviewers https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts 
- [ ] Implement coding guidelines and ensure code quality.
- [ ] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [ ] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com
- [ ] Adhere to readme template 
- [ ] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [ ] Enforce format via clang-format config file
- [ ] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [ ] Review readme with Tom Lenth or Joe Oster. (GitHub User: JoeOster)
- [ ] Tested using Dev Cloud when applicable
- [ ] Implement fixes for ONSAM Jiras
- [ ] If you have new dependencies/binaries, inform Samples Project Manager Swapna R Dontharaju (@srdontha) or @JoeOster

### CODE REVIEW

**Review by JoeOster — DISMISSED** [review]
approving for kicking off ci

File: DirectProgramming/DPC++/N-BodyMethods/Nbody/src/GSimulation.cpp [hunk 1]
```diff
@@ -158,8 +158,8 @@ void GSimulation::Start() {
     // Second kernel updates the velocity and position for all particles
     q.submit([&](handler& h) {
        auto p = pbuf.get_access<access::mode::read_write>(h);
```
Comment (pmpeter1) [inline DirectProgramming/DPC++/N-BodyMethods/Nbody/src/GSimulation.cpp:160]: Need new SYCL 2020 form of accessor.

**Review by pmpeter1 — CHANGES_REQUESTED** [review]

File: DirectProgramming/DPC++/N-BodyMethods/Nbody/src/GSimulation.cpp [hunk 2]
```diff
@@ -108,14 +104,17 @@ void GSimulation::Start() {
...
   // Create a queue to the selected device and enabled asynchronous exception
   // handling for that queue
   queue q(default_selector{}, dpc_common::exception_handler);
   // Create SYCL buffer for the Particle array of size "n"
   buffer pbuf(particles_.data(), r,
               {cl::sycl::property::buffer::use_host_ptr()});
-  // Create SYCL buffer for the ener array
-  buffer ebuf(energy.data(), r, {cl::sycl::property::buffer::use_host_ptr()});
+  // Allocate energy using USM allocator shared
+  RealType *energy = (RealType *)malloc_shared(sizeof(RealType),q);
```
[the remark this answers is not in the document]
Comment (mpanoop) [reply DirectProgramming/DPC++/N-BodyMethods/Nbody/src/GSimulation.cpp]: Fixed this in latest code

File: DirectProgramming/DPC++/N-BodyMethods/Nbody/src/GSimulation.cpp (hunk 1 above)
Comment (mpanoop) [reply DirectProgramming/DPC++/N-BodyMethods/Nbody/src/GSimulation.cpp]: Fixed this in latest code.

**Review by JoeOster — APPROVED** [review]
approving to launch ci

**Review by anjgola — APPROVED** [review]

**Pull request merged**